graphjet-core/src/main/java/com/twitter/graphjet/algorithms/counting/TopSecondDegreeByCountTweetRecsGenerator.java

```diff
@@ -83,30 +81,91 @@ private static boolean isTweetSocialProofOnly(
     SmallArrayBasedLongToDoubleMap[] socialProofs,
     int tweetSocialProofType
   ) {
-    boolean keep = false;
+    boolean lessThan = true;
```

Comment:
Good observation! As a matter of fact, I think this is true throughout this file and probably a better and cleaner pattern!

Can you please update all the occurrences in this file along your other changes?